From 73b302f253e8ec8e50a1a48abd6df971b92c534c Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Wed, 9 Mar 2011 09:51:41 -0800
Subject: [PATCH 072/236] MIPS: Octeon: cn6xxx platform config updates

Source: Cavium SDK 2.1.0-407

Configure cn68xx platform for one mgmt port and 4 MDIO busses
Configure cn66xx platform for two mgmt ports and 2 MDIO busses

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
Integrated-by: Phil Staub <Phil.Staub@windriver.com>
---
 arch/mips/cavium-octeon/octeon-platform.c |   10 +++++++---
 drivers/net/octeon/ethernet.c             |    3 ++-
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/arch/mips/cavium-octeon/octeon-platform.c b/arch/mips/cavium-octeon/octeon-platform.c
index 6089604..412f04b 100644
--- a/arch/mips/cavium-octeon/octeon-platform.c
+++ b/arch/mips/cavium-octeon/octeon-platform.c
@@ -202,7 +202,8 @@ static int __init octeon_i2c_device_init(void)
 	};
 
 	if (OCTEON_IS_MODEL(OCTEON_CN56XX) || OCTEON_IS_MODEL(OCTEON_CN52XX)
-	    || OCTEON_IS_MODEL(OCTEON_CN63XX))
+	    || OCTEON_IS_MODEL(OCTEON_CN63XX)
+	    || OCTEON_IS_MODEL(OCTEON_CN66XX))
 		num_ports = 2;
 	else
 		num_ports = 1;
@@ -272,8 +273,11 @@ static int __init octeon_mdiobus_device_init(void)
 		return 0;
 
 	if (OCTEON_IS_MODEL(OCTEON_CN56XX) || OCTEON_IS_MODEL(OCTEON_CN52XX)
-	    || OCTEON_IS_MODEL(OCTEON_CN63XX))
+	    || OCTEON_IS_MODEL(OCTEON_CN63XX)
+	    || OCTEON_IS_MODEL(OCTEON_CN66XX))
 		num_mdio = 2;
+	else if (OCTEON_IS_MODEL(OCTEON_CN68XX))
+		num_mdio = 4;
 	else
 		num_mdio = 1;
 
@@ -318,7 +322,7 @@ static int __init octeon_mgmt_device_init(void)
 	    && !OCTEON_IS_MODEL(OCTEON_CN6XXX))
 		return 0;
 
-	if (OCTEON_IS_MODEL(OCTEON_CN56XX))
+	if (OCTEON_IS_MODEL(OCTEON_CN56XX) || OCTEON_IS_MODEL(OCTEON_CN68XX))
 		num_ports = 1;
 	else
 		num_ports = 2;
diff --git a/drivers/net/octeon/ethernet.c b/drivers/net/octeon/ethernet.c
index 8413376..dd5ec45 100644
--- a/drivers/net/octeon/ethernet.c
+++ b/drivers/net/octeon/ethernet.c
@@ -749,7 +749,8 @@ static int __init cvm_oct_init_module(void)
 
 	cvmx_override_pko_queue_priority = cvm_oct_override_pko_queue_priority;
 
-	if (OCTEON_IS_MODEL(OCTEON_CN52XX) || OCTEON_IS_MODEL(OCTEON_CN63XX))
+	if (OCTEON_IS_MODEL(OCTEON_CN52XX) || OCTEON_IS_MODEL(OCTEON_CN63XX)
+	    || OCTEON_IS_MODEL(OCTEON_CN66XX))
 		cvm_oct_mac_addr_offset = 2; /* First two are the mgmt ports. */
 	else if (OCTEON_IS_MODEL(OCTEON_CN56XX) || OCTEON_IS_MODEL(OCTEON_CN68XX))
 		cvm_oct_mac_addr_offset = 1; /* First one is the mgmt port. */
-- 
1.7.0

